packages/shared/components/MenuItem.svelte

```diff
@@ -1,44 +1,52 @@
 <script lang="ts">
     import { Icon, Text, Spinner, Position, InformationTooltip } from 'shared/components'
+    import { Icon as IconEnum } from '@auxiliary/icon'
+    import { MenuItemVariant } from './enums'
 
-    export let icon: string = ''
-    export let iconProps: Record<string, unknown> = undefined
+    export let icon: IconEnum | undefined = undefined
+    export let iconProps: Record<string, unknown> | undefined = undefined
     export let title: string
-    export let subtitle = ''
-    export let onClick: () => any
-    export let selected = false
-    export let disabled = false
-    export let isLoading = false
-    export let enableTooltipVisible = false
-    export let tooltip: string = undefined
+    export let subtitle: string = ''
+    export let onClick: () => void
+    export let selected: boolean = false
+    export let disabled: boolean = false
+    export let isLoading: boolean = false
+    export let enableTooltipVisible: boolean = false
+    export let tooltip: string | undefined = undefined
+    export let variant: MenuItemVariant = MenuItemVariant.Info
 
-    export let variant: 'success' | 'error' | 'warning' | 'info' = 'info'
+    enum Color {
+        Blue = 'blue',
+        Green = 'green',
+        Yellow = 'yellow',
+        Red = 'red',
+    }
 
     let showTooltip = false
     let menuItem: HTMLElement
-    let color = 'blue'
+    let color = Color.Blue
```

Comment:
instead of this and `$: variant, setColor()` it's better to just make a colorMap type `Record<MenuItemVariant, string>` and do `$: color = colorMap[variant]`